test/StdUtils.t.sol

```diff
@@ -74,6 +74,76 @@ contract StdUtilsTest is Test {
         bound(num, min, max);
     }
 
+    function testBoundInt_DD() public {
+        assertEq(boundInt(-3, 0, 4), 2);
+        assertEq(boundInt(0, -69, -69), -69);
+        assertEq(boundInt(0, -69, -68), -68);
+        assertEq(boundInt(-10, 150, 190), 154);
+        assertEq(boundInt(-300, 2800, 3200), 2908);
+        assertEq(boundInt(9999, -1337, 6666), 1995);
+    }
+
+    function testBoundInt_WithinRange() public {
+        assertEq(boundInt(51, -50, 150), 51);
+        assertEq(boundInt(51, -50, 150), boundInt(boundInt(51, -50, 150), -50, 150));
+        assertEq(boundInt(149, -50, 150), 149);
+        assertEq(boundInt(149, -50, 150), boundInt(boundInt(149, -50, 150), -50, 150));
+    }
+
+    function testBoundInt_EdgeCoverage() public {
+        assertEq(boundInt(type(int256).min, -50, 150), -50);
+        assertEq(boundInt(type(int256).min + 1, -50, 150), -49);
+        assertEq(boundInt(type(int256).min + 2, -50, 150), -48);
+        assertEq(boundInt(type(int256).min + 3, -50, 150), -47);
+        assertEq(boundInt(type(int256).max, -50, 150), 150);
+        assertEq(boundInt(type(int256).max - 1, -50, 150), 149);
+        assertEq(boundInt(type(int256).max - 2, -50, 150), 148);
+        assertEq(boundInt(type(int256).max - 3, -50, 150), 147);
```

Comment:
Can we also tests for the inverse cases, e.g.
```solidity
assertEq(boundInt(type(int256).min, 10, 150), 10);
assertEq(boundInt(type(int256).min + 1, 10, 150), 11);
// add the other two
assertEq(boundInt(type(int256).max, -50, -10), -10);
assertEq(boundInt(type(int256).max - 1, -50, -10), -11);
// add the other two
```

Maybe good to explicitly test when zero is both the lower and upper bound in those scenarios also. Ints have some more edge cases than Uints IME so just want to make sure it's well tested 🙂

Reply:
Great! I'll add them.

Reply:
[c43e95b](https://github.com/foundry-rs/forge-std/pull/253/commits/c43e95be0f0771d0d188cd1b855ad0fe972b5b5c)

Reply:
Small nit, do you mind reordering the tests like this? It makes it easier to read since sequential/related tests are next to each other. (I'm happy to push the commit, but don't want to push yet in case you're working on https://github.com/foundry-rs/forge-std/pull/253#discussion_r1045104818). 

<img width=500 src="https://user-images.githubusercontent.com/17163988/206866693-86986608-1a21-4424-8532-e49474b3fc12.png"/>

Reply:
Got it, I'll apply it.

Reply:
Fixed at [here](https://github.com/foundry-rs/forge-std/pull/253/commits/3e50c5d244cdfa055aa6468334ad6193e746d7fe).